src/AutSoft.AspNetCore.Blazor/Loading/LoadingOperation.cs

```diff
@@ -0,0 +1,215 @@
+using AutSoft.AspNetCore.Blazor.Clipboard;
+using AutSoft.AspNetCore.Blazor.ErrorHandling;
+
+using Microsoft.Extensions.Logging;
+
+using MudBlazor;
+
+namespace AutSoft.AspNetCore.Blazor.Loading;
+
+/// <summary>
+/// Loading operation.
+/// </summary>
+public class LoadingOperation
+{
+    private readonly ILogger _logger;
+    private readonly IDisplayErrorFactory _displayErrorFactory;
+    private readonly DefaultLoadingErrorHandlerFactory _defaultLoadingErrorHandlerFactory;
+    private readonly IClipboardService _clipboardService;
+    private readonly ISnackbar _snackbar;
+
+    private Func<Task> _currentBody = () => Task.CompletedTask;
+    private ILoadingErrorHandler? _currentErrorHandler;
+
+    /// <summary>
+    /// Loading operation state.
+    /// </summary>
+    [System.Diagnostics.CodeAnalysis.SuppressMessage("Design", "CA1051:Do not declare visible instance fields", Justification = "Base service")]
+    [System.Diagnostics.CodeAnalysis.SuppressMessage("Style", "IDE1006:Naming Styles", Justification = "Base service")]
+    protected LoadingStateType _state = LoadingStateType.Done;
```

Comment:
propertyk legyen a protected tagok

Reply:
De ezen tagokat propertyk-hez használjuk az nem gond?

Reply:
de ha a property protected, akkor a leszármazott azon keresztül el tudja érni, és ennek privátnak kellene lennie

Reply:
Igen, vannak is hozzá public-os property-k, amin keresztül a leszármazottak elérnék, úgyhogy ezeket a tagokat akkor private láthatóságúvá teszem és a leszármazottakban átírom, hogy a propertyket használják